src/js/components/agencyV2/statusOfFunds/StatusOfFunds.jsx

```diff
@@ -72,13 +75,56 @@ const StatusOfFunds = ({ fy }) => {
             });
     };
 
+    const fetchFederalAccounts = (agencyData) => {
+        if (request.current) {
+            request.current.cancel();
+        }
+        if (error) {
+            setError(false);
+        }
+        if (!loading) {
+            setLoading(true);
+        }
+        const params = {
+            limit: pageSize,
+            page: currentPage
+        };
+        request.current = fetchFederalAccountsList(overview.toptierCode, agencyData.id, fy, params.page);
+        const federalAccountsRequest = request.current;
+        federalAccountsRequest.promise
+            .then((res) => {
+                const parsedData = parseRows(res.data.results);
+                const totalsData = {
+                    name: `${agencyData.name}`,
+                    id: `${agencyData.id}`,
+                    total_budgetary_resources: `${agencyData.budgetaryResources}`,
+                    total_obligations: `${agencyData.obligations}`
+                };
+                setLevel(1, totalsData);
+                setResults(parsedData);
+                dispatch(setFederalAccountsList(parsedData));
+                setTotalItems(res.data.page_metadata.total);
+                setLoading(false);
+            }).catch((err) => {
+                setError(true);
+                setLoading(false);
+                console.error(err);
+            });
+    };
+
+    useEffect(() => {
+        if (Object.keys(subcomponent).length !== 0) {
+            fetchFederalAccounts(subcomponent);
+        }
+    }, [subcomponent]);
+
     useEffect(() => {
-        const hasParamChanged = (
-            prevPage !== currentPage || prevPageSize !== pageSize
-        );
-        if (hasParamChanged) {
+        if (prevPage !== currentPage && level === 0) {
```

Comment:
If you add the following condition to prevent fetching new data in this useEffect when !(prevPage > 1 && currentPage === 1) and add setLoading(true) in the onClick function that fixes the issues. I'm not sure if this works for all cases and it seems like there should be a better way to check !(prevPage > 1 && currentPage === 1) such as setting a new state to transitioning and to prevent fetching new data here when the transitioning state is set.

Reply:
Thanks Andrea I pushed a fix adding the check for currentPage !== 1.  I double checked the tab/enter functionality and it is still working on my end, the tab through starts with the x axis labels then goes through each bar group- are you not able to tab through the chart at all? @anjenkin

Reply:
I see tabbing works.  I was expecting a different tab order. I was thinking you could tab to the label and then the next tab would set focus to the bar next to the label.

There's an issue with checking currentPage !== 1 because that breaks pagination if you select a page greater than 1 and then going back to page 1.

I just tested the following and it works if you set state while the chart is resetting in this case.  So in the useEffect that responses to the currentPage changing then if you can check for the "resetting state" and don't fetch the data if resetting.  This is a tricky visualization!


```
    const [resetBarCnt, setResetBarCnt] = useState(false);

    useEffect(() => {
        if (resetBarCnt) {
            setResetBarCnt(false);
        } else {
            if (prevPage !== currentPage && level === 0) {
                fetchAgencySubcomponents();
            }
            if (prevPage !== currentPage && level === 1) {
                fetchFederalAccounts(subcomponent);
            }
        }
    }, [currentPage]);

    useEffect(() => {
        if (resetBarCnt) {
            setLoading(true);
            changeCurrentPage(1);
        }
    }, [resetBarCnt]);

    const onClick = (selectedLevel, data) => {
        // reset to page 1 on drilldown
        setResetBarCnt(true);
        const subcomponentTotalData = Object.create(BaseStatusOfFundsLevel);
        subcomponentTotalData.populate(data);
        dispatch(setSelectedSubcomponent(subcomponentTotalData));
        setSubcomponent(subcomponentTotalData);
    };
```

Reply:
Ah thank you good catch on the behavior when changing to page 1 from another page > 1 I had missed that condition when testing - I updated the PR with your page change state logic.
Re: tabbing, I have set a tab-index of 0 to both the bar groups and y axis labels so the tab indexes are grouped by d3 element

Reply:
Geez, I found one more issue while testing. If you click on a sub-component on page 1 (which doesn't trigger a page change), click back, and then try to paginate, the data doesn't update because we didn't trigger a page change and the resetPageChange state isn't cleared.  I think we should be good just setting resetPageChange state to false in the goBack function.  I can't think of another case where we would have an issue.

Reply:
Ahh setting page change to false in onClick resolved both issues for me - just pushed updates let me know if you are still seeing the issue

Reply:
Yeah this seems to work - 
`    useEffect(() => {
        if (resetPageChange) {
            setLoading(true);
            if (currentPage === 1) {
                setResetPageChange(false);
            } else {
                changeCurrentPage(1);
            }
        }
    }, [resetPageChange]);

    useEffect(() => {
        if (fy && overview.toptierCode) {
            fetchAgencySubcomponents();
        }
    }, [fy, overview.toptierCode]);

    const onClick = (selectedLevel, data) => {
        // reset to page 1 on drilldown
        setResetPageChange(true);
        const subcomponentTotalData = Object.create(BaseStatusOfFundsLevel);
        subcomponentTotalData.populate(data);
        dispatch(setSelectedSubcomponent(subcomponentTotalData));
        setSubcomponent(subcomponentTotalData);
    };`

Reply:
Thanks this seems to work on my end too - I included the same logic to resolve a flickering issue with the goBack() which fixed that as well.. fingers crossed that this is the golden fix 🙏

Reply:
This works but if I go to the Dept of Treasury V2 page, click the Departmental Offices sub-component and go to page 2, the bars are the wrong height and it's only showing 5 bars.  However, if I go to page 3 and 4, the bars show correctly
<img width="1079" alt="Screen Shot 2022-01-11 at 12 18 47 PM" src="https://user-images.githubusercontent.com/121010/148990461-fd48aa4f-89d8-47d8-b188-f6b4babb243c.png">
.

Reply:
@anjenkin Thanks for that catch! The way we were truncating texts caused some y axis labels to have the same value which causes d3 to put them in the same place. Just pushed an update so that we are now doing the truncation after getting the full subcomponent name so this resolves the bar heights issue